go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -539,21 +539,44 @@ func (p DoltDatabaseProvider) DropDatabase(ctx *sql.Context, name string) error
 	dbKey := formatDbMapKeyName(name)
 	db := p.databases[dbKey]
 
-	// Get the DB's directory
-	exists, isDir := p.fs.Exists(db.Name())
-	if !exists {
-		// engine should already protect against this
+	// get location of database that's being dropped
+	dbLoc := p.dbLocations[dbKey]
+	if dbLoc == nil {
 		return sql.ErrDatabaseNotFound.New(db.Name())
-	} else if !isDir {
-		return fmt.Errorf("unexpected error: %s exists but is not a directory", dbKey)
 	}
-
-	err = p.fs.Delete(db.Name(), true)
+	dropDbLoc, err := dbLoc.Abs("")
+	if err != nil {
+		return err
+	}
+	currentDbLoc, err := p.fs.Abs("")
 	if err != nil {
 		return err
 	}
+	dirToDelete := ""
+	// if the database is in the directory itself, we remove '.dolt' directory rather than
+	// the whole directory itself because it can have other databases that are nested.
```

Comment:
Excellent comment! This is a very important point, so good use of a comment to make sure it's clear to all future readers. Deleting other databases a user did not request would be a very bad mistake. 

Speaking of... this would be a really good thing for us to test carefully, too... for example... have a test set up that has a database directly in the db root directory as well as nested db directories, delete the root db, and then do some sanity query like `show databases` or something to make sure the other databases are still present.